docs/docsite/rst/dev_guide/developing_modules_general_windows.rst

```diff
@@ -166,17 +166,17 @@ Windows new module development
 When creating a new module there are a few things to keep in mind:
 
 - Module code is in Powershell (.ps1) files while the documentation is contained in Python (.py) files of the same name
-- Avoid using ``Write-Host/Debug/Verbose/Error`` in the module and add what needs to be returned to the ``$result`` variable
-- When trying an exception use ``Fail-Json -obj $result -message "exception message here"`` instead
+- Avoid using ``Write-Host/Debug/Verbose/Error`` in the module and add what needs to be returned to the ``$module.Result`` variable
+- To fail a module, call ``$module.FailJson("failure message here")``, an Exception or ErrorRecord can be set to the second argument for a more descriptive error message
 - Most new modules require check mode and integration tests before they are merged into the main Ansible codebase
 - Avoid using try/catch statements over a large code block, rather use them for individual calls so the error message can be more descriptive
 - Try and catch specific exceptions when using try/catch statements
 - Avoid using PSCustomObjects unless necessary
 - Look for common functions in ``./lib/ansible/module_utils/powershell/`` and use the code there instead of duplicating work. These can be imported by adding the line ``#Requires -Module *`` where * is the filename to import, and will be automatically included with the module code sent to the Windows target when run via Ansible
+- As well as PowerShell module utils, C# module utils are stored in ``./lib/ansible/module_utils/csharp/`` and are automatically imported in a module execution if the line ``#AnsibleRequires -CSharpUtil *`` is present
```

Comment:
Ah, this answers my question above - C# module utils are not replacing the PowerShell module utils, just adding to them. Would a developer choose only one, or could s/he use both?

Reply:
I'll add a bit around the choices between the 2.